_posts/2020-05-02-schedule-your-functions.md

```diff
@@ -0,0 +1,264 @@
+
+Now we need our functions to to be invoked on an interval. In Linux we have the [cron](https://en.wikipedia.org/wiki/Cron) that allows us to execute code on a schedule. In Kubernetes, this was abstracted into the `CronJob` object.  For any OpenFaaS cluster, the [cron-connector](https://github.com/openfaas-incubator/cron-connector) uses the `connector-sdk` to provide the same functionality, running our functions on a predefined schedule.
+
+Now lets roll up our sleeves and see how we can execute our OpenFaaS functions on time.
+
+## Prerequisites
+
+Before we start we need a couple of tools to help us quickly set up our environment:
+
+* [docker](https://docs.docker.com/get-docker/) - the container runtime used in this post
+* [kind](https://kind.sigs.k8s.io/docs/user/quick-start/) - lightweight local Kubernetes cluster which uses containers as nodes
+* [arkade](https://github.com/alexellis/arkade) - one line installation of applications with which we will install OpenFaaS
+* [faas-cli](https://docs.openfaas.com/cli/install/) - the CLI which communicates with the OpenFaaS gateway
+* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) - the CLI which communicates with your Kubernetes cluster
+
+## Prepare environment
+
+Before showing you how to run our functions on a schedule, first we need to set up the environment with the tools mentioned above.
+
+### Create cluster
+
+First thing we need is running Kubernetes cluster:
+
+```bash
+kind create cluster
+```
+
+Wait for the installation to finish and verify the pods in the `kube-system` namespace are `1/1`:
+
+```bash
+kubectl get pods -n kube-system
+```
+
+### Install OpenFaaS
+
+With `arkade` the installation of OpenFaaS boils down to single line command:
+
+```bash
+arkade install openfaas
+```
+
+Wait for the OpenFaaS gateway to be ready:
+
+```bash
+kubectl rollout status -n openfaas deploy/gateway
+```
+
+### Set up OpenFaaS client
+
+Get the OpenFaaS Gateway password:
+
+```bash
+PASSWORD=$(kubectl get secret -n openfaas basic-auth -o jsonpath="{.data.basic-auth-password}" | base64 --decode; echo)
+```
+
+Set up the gateway endpoint for the client:
+
+```bash
+export OPENFAAS_URL=http://$(docker inspect -f "{{ .NetworkSettings.IPAddress }}" $(docker ps -aqf "name=kind-control-plane")):31112
+```
+
+Authenticate the client:
+
+```bash
+echo -n $PASSWORD | faas-cli login --username admin --password-stdin
+```
+
+## Schedule the functions
+
+After we have the environment up and running, now we can install the cron-connector and schedule our functions to run at specific intervals.
+
+### Deploy the cron-connector
+
+In order to install our `cron-connector` and schedule a function run the following command:
+
+```bash
+arkade install cron-connector
+```
+
+Wait for the `cron-connector` to be deployed:
+
+```bash
+kubectl rollout status -n openfaas deploy/sample-connector
+```
+
+## How to schedule
+
```

Comment:
Please include or one two examples of cron schedules in bullets:

* every 5 minutes
* every day
* every week

Reply:
Good idea :+1:

Reply:
Added the proposed examples :heavy_check_mark: